Windows/MainWindow.xaml.cs

```diff
@@ -69,6 +69,7 @@ public partial class MainWindow : Window {
         bool shiftKeyDown;
         bool ctrlKeyDown;
         bool returnToStartMenuOnClose = false;
+        bool doneInit = false;
```

Comment:
I feel like there's a better way we can implement this.

`doneInit` is only set in two functions, right?

Maybe store the dirtiness state of the map before initialisation (i.e. when `doneInit` is set to false), and then afterwards (when `doneInit` is set to true) restore the previous state.

That should stop the map getting marked as dirty by non-user changes.

Reply:
Done - had to check which state actually was mutated by those initialization to avoid writing too much logic, but it works.

By the way, my change with ```doneInit``` actually had introduced a regression with Recently Viewed songs that I didn't notice before - they weren't updated if you opened a map and immediately closed it, so it's good that this was refactored.